api/fossa/tar.go

```diff
@@ -64,7 +64,7 @@ func UploadTarballProject(dir string, rawLicenseScan bool) (Locator, error) {
 // Since this will be running within CI machines, this is probably not a good
 // idea. (See https://circleci.com/docs/2.0/configuration-reference/#resource_class
 // for an example of our memory constraints.)
-func UploadTarball(name string, dir string, dependency, rawLicenseScan, upload bool) (Locator, error) {
+func UploadTarball(name, revision, dir string, dependency, rawLicenseScan, upload bool, uploadOptions UploadOptions) (Locator, error) {
```

Comment:
Non-blocking given urgency, but it would be better style here for `UploadTarball` to take an options struct e.g.

```go
type UploadTarballOptions struct {
  Name string
  Revision string
  // ...
}
```

This helps people avoid accidentally mix up the order of arguments with the same type (e.g. `name` and `revision`, or `rawLicenseScan` and `upload`). Besides, you're taking an `UploadOptions` struct anyway.

Reply:
Yea I like this, I'll implement this here.